auth/src/ios/credential_ios.mm

```diff
@@ -166,9 +166,16 @@ @implementation PhoneListenerDataObjC
   FIREBASE_ASSERT(future_api != nullptr);
 
   const auto handle = future_api->SafeAlloc<Credential>(kCredentialFn_GameCenterGetCredential);
+  /**
+   Linking GameKit.framework without using it on macOS results in App Store rejection.
+   Thus we don't link GameKit.framework to our SDK directly. `optionalLocalPlayer` is used for
+   checking whether the APP that consuming our SDK has linked GameKit.framework. If not, will
+   complete with kAuthErrorInvalidCredential error.
+   **/
+  GKLocalPlayer *_Nullable optionalLocalPlayer = [[NSClassFromString(@"GKLocalPlayer") alloc] init];
```

Comment:
By referencing `GKLocalPlayer`, the linker will try to link to Gamekit.framework.  We should avoid using this class entirely. (As well as `#import <GameKit/GameKit.h>` at the top of this file)

I think we can just use [NSClassFromString](https://developer.apple.com/documentation/foundation/1395135-nsclassfromstring). Ex.
```
if (NSClassFromString(@"GKLocalPlayer" == nil) {
  return false
}
```

Similar to the change below

Reply:
Similarly, since you are following https://github.com/firebase/firebase-ios-sdk/pull/2355/files, this should be fine